### PR TITLE
[Fixes #6591] TempData should support nullable types

### LINKED ISSUE #6591: TempData should support nullable types
When binding a property through temp data nullable types should be supported as a means to represent data that might not be there.

For example, if I want to detect if some primitive value was not part of the temp data cause I arrived at the page through some other flow, I want to be able to check for null and do something else instead.

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/TempDataApplicationModelProviderTest.cs [hunk 1]
```diff
@@ -108,6 +109,36 @@ public void ThrowsInvalidOperationException_NonPrimitiveType()
...
+
+            Assert.Equal($"The '{controllerType.FullName}.{nameof(TestController_NullableNonPrimitiveTempDataProperty.Car)}'"
+                + $" property with {nameof(TempDataAttribute)} is invalid. A property using {nameof(TempDataAttribute)} "
+                + $"must be of primitive or string type.", exception.Message);
+        }
+
+        public class TestController_NullableNonPrimitiveTempDataProperty
+        {
+            [TempData]
+            public Car? Car { get; set; }
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/TempDataApplicationModelProviderTest.cs]: Could replace it with a BCL type like `DateTime`

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/TempDataApplicationModelProviderTest.cs [hunk 2]
```diff
@@ -108,6 +109,36 @@ public void ThrowsInvalidOperationException_NonPrimitiveType()
...
+            var context = new ApplicationModelProviderContext(new[] { controllerType.GetTypeInfo() });
+            defaultProvider.OnProvidersExecuting(context);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() =>
+                provider.OnProvidersExecuting(context));
+
+            Assert.Equal($"The '{controllerType.FullName}.{nameof(TestController_NullableNonPrimitiveTempDataProperty.Car)}'"
+                + $" property with {nameof(TempDataAttribute)} is invalid. A property using {nameof(TempDataAttribute)} "
+                + $"must be of primitive or string type.", exception.Message);
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/TempDataApplicationModelProviderTest.cs]: should the message say 

> must be ~~of~~ a primitive or string type

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataPropertyTest.cs [hunk 3]
```diff
@@ -25,13 +25,13 @@ public TempDataPropertyTest(MvcTestFixture<BasicWebSite.Startup> fixture)
         public async Task TempDataPropertyAttribute_RetainsTempDataWithView()
         {
             // Arrange
-            var message = "Success (from Temp Data)";
+            var tempDataContent = "Success (from Temp Data)100";
```
Comment (javiercn) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataPropertyTest.cs]: Why do you need that 100 at the end?

Comment (kichalla) [reply test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataPropertyTest.cs]: Its just a response message that we are trying to verify in the test for the following action:
https://github.com/aspnet/Mvc/pull/7102/files#diff-ed958851cdea5a1bf43528eba0262c0bR57

**Pull request merged**